src/openms/include/OpenMS/KERNEL/MSExperiment.h

```diff
@@ -336,62 +337,180 @@ namespace OpenMS
     /// Returns a non-mutable area iterator for all peaks in @p range. If a dimension is empty(), it is ignored (i.e. does not restrict the area)
     ConstAreaIterator areaBeginConst(const RangeManagerType& range, UInt ms_level = 1) const;
 
-    /// Returns an non-mutable invalid area iterator marking the end of an area
+    /// Returns a non-mutable invalid area iterator marking the end of an area
     ConstAreaIterator areaEndConst() const;
 
-    // for fast pyOpenMS access to MS1 peak data in format: [rt, [mz, intensity]]
-    void get2DPeakData(CoordinateType min_rt, CoordinateType max_rt, CoordinateType min_mz, CoordinateType max_mz, 
+    /* @brief Retrieves the peak data in the given mz-rt range and store data spectrum-wise in separate arrays.
+     * 
+     * For fast pyOpenMS access to peak data in format: [rt, [mz, intensity]]
+     * 
+     * @param min_rt The minimum retention time.
+     * @param max_rt The maximum retention time.
+     * @param min_mz The minimum m/z value.
+     * @param max_mz The maximum m/z value.
+     * @param ms_level The MS level of the spectra to consider.
+     * @param rt The vector to store the retention times in.
+     * @param mz The vector to store the m/z values in.
+     * @param intensity The vector to store the intensities in.
+     */
+    void get2DPeakDataPerSpectrum(
+      CoordinateType min_rt, 
+      CoordinateType max_rt, 
+      CoordinateType min_mz, 
+      CoordinateType max_mz,
```

Comment:
A lot of CoordinateType objects in a row might be confusing and could be mixed up. Maybe the Range object should be used instead?

Reply:
True. I have no strong opinion, but it will be an additional step to construct those object in python.

I mostly based it on the existing functions. We should change them all if we agree on using Range objects anywhere.

Reply:
I am a bit divided here. So we would have different range classes wrapped?

Reply:
We kind of have it already or at least will need it for our multi extraction.

RangeMZ(double, double) and RangeRt(double, double) would be how to construct them.

Reply:
K leave this for next iteration as not really urgent